stores/rocksdb/src/commonMain/kotlin/kotlinx/document/database/rocksdb/RocksdbDataStore.kt

```diff
@@ -1,23 +1,66 @@
 package kotlinx.document.database.rocksdb
 
+import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.IO
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 import kotlinx.document.database.DataStore
 import kotlinx.document.database.PersistentMap
 import maryk.rocksdb.Options
 import maryk.rocksdb.RocksDB
+import maryk.rocksdb.WriteOptions
 import maryk.rocksdb.openRocksDB
 import maryk.rocksdb.use
+import kotlin.coroutines.CoroutineContext
+
+public class RocksdbDataStore(
+    internal val delegate: RocksDB,
+    override val configuration: DataStore.Configuration,
+) : DataStore, CoroutineScope {
+    override val coroutineContext: CoroutineContext by lazy { SupervisorJob() + Dispatchers.IO }
+
+    init {
+        when (val policy = configuration.commitPolicy) {
+            is DataStore.CommitPolicy.Periodic -> {
+                launch {
+                    while (true) {
+                        delay(policy.interval)
+                        commit()
+                    }
+                }
+            }
+
+            else -> {}
+        }
+    }
 
-public class RocksdbDataStore(private val delegate: RocksDB) : DataStore {
     public companion object {
-        public fun open(path: String): RocksdbDataStore = RocksdbDataStore(openRocksDB(path))
+        public fun open(configuration: DataStore.Configuration): RocksdbDataStore = open(Options().setCreateIfMissing(true), configuration)
 
         public fun open(
             options: Options,
-            path: String,
-        ): RocksdbDataStore = RocksdbDataStore(openRocksDB(options, path))
+            configuration: DataStore.Configuration,
+        ): RocksdbDataStore =
+            RocksdbDataStore(
+                delegate = openRocksDB(options = options, path = configuration.path.toString()),
+                configuration = configuration,
+            ).also {
```

Comment:
Do not use an `also` for such a long thing to write, it won't make the code more concise or shorter. Convert `open` body into a block and write it imperatively here, It would be much more readable.